src/data/index.ts

```diff
@@ -0,0 +1,49 @@
+import { configureStore, Dispatch } from '@reduxjs/toolkit';
+import { persistReducer, persistStore } from 'redux-persist';
+// import loggerMiddleware from 'redux-logger';
+import MementoStorage from './storage';
+
+import rootReducer, { actions } from './slice';
+import { Memento } from 'vscode';
+
+const buildStore = (workspaceState: Memento) => {
+	const persistConfig = {
+		key: 'root',
+		storage: new MementoStorage(workspaceState),
+		// debugger: console.log,
+	};
+
+	const persistedReducer = persistReducer(persistConfig, rootReducer);
+
+	// @TODO ensure that building with NODE_ENV=production var, otherwise default dev middleware will be included
```

Comment:
did you figure out how that one should work with our current build system?

Reply:
okay, i need to fix package.json for that